### ISSUE TITLE
Command error loading SNMP config

### ISSUE THREAD

**Issue by willtome** [ansible-network/cisco_ios#62]
Error when trying to load SNMP config lines with config manager. Devices are Cisco CSR running in AWS

```playbook.yml
---
- hosts: all
  gather_facts: no
  connection: network_cli
  vars:
    - config_manager_replace: no
    - config_blocks:
        - snmp_config
 
  tasks:
  - include_role:
      name: auto_config

  - debug:
      msg: "{{ configuration }}"

  - include_role:
      name: ansible-network.config_manager
      tasks_from: load
    vars:  
      config_manager_text: "{{ configuration }}"
```

```log.txt

PLAY [all] *************************************************************************************************************************************************************************************************

TASK [include_role : auto_config] **************************************************************************************************************************************************************************

TASK [auto_config : create config] *************************************************************************************************************************************************************************
ok: [rtr1] => (item=snmp_config)
ok: [rtr2] => (item=snmp_config)

TASK [auto_config : store config] **************************************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [debug] ***********************************************************************************************************************************************************************************************
ok: [rtr1] => {
    "msg": "snmp-server community ansible-public RO\nsnmp-server community ansible-private RW\n!\n"
}
ok: [rtr2] => {
    "msg": "snmp-server community ansible-public RO\nsnmp-server community ansible-private RW\n!\n"
}

TASK [include_role : ansible-network.config_manager] *******************************************************************************************************************************************************

TASK [ansible-network.config_manager : validate role spec] *************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.config_manager : create temp working directory] **************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : checkout scm project] ***********************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : discover the configuration file path and load it] *******************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : set the config_manager_file value based on config_manager_scm_file] *************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : remove temporary working dir] ***************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : load config file contents] ******************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : validate config_manager_text is defined] ****************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [invoke network provider] *****************************************************************************************************************************************************************************

TASK [ansible-network.cisco_ios : initialize function] *****************************************************************************************************************************************************
included: /Users/wtome/Documents/customers/Marriott/network-poc/roles/ansible-network.cisco_ios/includes/init.yaml for rtr1, rtr2

TASK [ansible-network.cisco_ios : set role basic facts] ****************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : display the role version to stdout] **************************************************************************************************************************************
ok: [rtr1] => {
    "msg": "ansible_network.cisco_ios version is v2.6.2"
}
ok: [rtr2] => {
    "msg": "ansible_network.cisco_ios version is v2.6.2"
}

TASK [ansible-network.cisco_ios : validate ansible_network_os == 'ios'] ************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : validate ansible_connection == 'network_cli'] ****************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : Validate we have required installed version of dependent roles] **********************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : Validate we have required installed version of dependent roles from meta] ************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : validate config_manager_text is defined] *********************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : set ios checkpoint filename] *********************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : create checkpoint of current configuration] ******************************************************************************************************************************
included: /Users/wtome/Documents/customers/Marriott/network-poc/roles/ansible-network.cisco_ios/includes/checkpoint/create.yaml for rtr1, rtr2

TASK [ansible-network.cisco_ios : validate ios_checkpoint_filename is defined] *****************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : get current files on disk] ***********************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : remove old checkpoint file (if necessary)] *******************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : create a checkpoint of the current running-config] ***********************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : display message due to check mode] ***************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : replace current active configuration] ************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : merge with current active configuration] *********************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : load configuration using configure terminal] *****************************************************************************************************************************
included: /Users/wtome/Documents/customers/Marriott/network-poc/roles/ansible-network.cisco_ios/includes/configure/terminal.yaml for rtr1, rtr2

TASK [ansible-network.cisco_ios : enter configuration mode] ************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : load configuration lines into target device] *****************************************************************************************************************************
ok: [rtr1] => (item=snmp-server community ansible-public RO)
ok: [rtr2] => (item=snmp-server community ansible-public RO)
ok: [rtr1] => (item=snmp-server community ansible-private RW)
ok: [rtr1] => (item=!)
fatal: [rtr1]: FAILED! => {"msg": "missing required argument `command`"}
ok: [rtr2] => (item=snmp-server community ansible-private RW)
ok: [rtr2] => (item=!)
fatal: [rtr2]: FAILED! => {"msg": "missing required argument `command`"}

TASK [ansible-network.cisco_ios : exit configuration mode] *************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : set host failed] *********************************************************************************************************************************************************
fatal: [rtr1]: FAILED! => {"changed": false, "msg": "error loading configuration lines"}
fatal: [rtr2]: FAILED! => {"changed": false, "msg": "error loading configuration lines"}

TASK [ansible-network.cisco_ios : display message] *********************************************************************************************************************************************************
ok: [rtr1] => {
    "msg": "error configuring device, starting rollback"
}
ok: [rtr2] => {
    "msg": "error configuring device, starting rollback"
}

TASK [ansible-network.cisco_ios : initiate configuration rollback] *****************************************************************************************************************************************
included: /Users/wtome/Documents/customers/Marriott/network-poc/roles/ansible-network.cisco_ios/includes/checkpoint/restore.yaml for rtr1, rtr2

TASK [ansible-network.cisco_ios : validate ios_checkpoint_filename is defined] *****************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : get current files on disk] ***********************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : verify checkpoint file exists] *******************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : checkpoint configuration restore pre hook] *******************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : restore checkpoint configuration] ****************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : checkpoint configuration restore post hook] ******************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : remove checkpoint file from remote device] *******************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : display message] *********************************************************************************************************************************************************
ok: [rtr1] => {
    "msg": "successfully completed configuration rollback"
}
ok: [rtr2] => {
    "msg": "successfully completed configuration rollback"
}

TASK [ansible-network.cisco_ios : fail host due to config load error] **************************************************************************************************************************************
fatal: [rtr1]: FAILED! => {"changed": false, "msg": "error loading configuration onto target device"}
fatal: [rtr2]: FAILED! => {"changed": false, "msg": "error loading configuration onto target device"}

PLAY RECAP *************************************************************************************************************************************************************************************************
rtr1                       : ok=21   changed=0    unreachable=0    failed=3   
rtr2                       : ok=21   changed=0    unreachable=0    failed=3   


```

**Comment by gdpak** [comment]
@willtome 
Here looks like '!' in config line failed. Not sure why ..
```
ok: [rtr1] => (item=!)
fatal: [rtr1]: FAILED! => {"msg": "missing required argument `command`"}
```
When I tried to play same line on my CSRv it did not complain . Can you try on your ASR manually what happens when you try below -
```
foo1(config)#!
foo1(config)#end
```

**Comment by willtome** [comment]
@gdpak I am also running against CSRv. Entering the line in manually does not cause an error

**Comment by willtome** [comment]
removing the `!` still causes the same error
```
PLAY [all] *************************************************************************************************************************************************************************************************

TASK [include_role : auto_config] **************************************************************************************************************************************************************************

TASK [auto_config : create config] *************************************************************************************************************************************************************************
ok: [rtr2] => (item=snmp_config)
ok: [rtr1] => (item=snmp_config)

TASK [auto_config : store config] **************************************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [debug] ***********************************************************************************************************************************************************************************************
ok: [rtr2] => {
    "msg": "snmp-server community ansible-public RO\nsnmp-server community ansible-private RW\n"
}
ok: [rtr1] => {
    "msg": "snmp-server community ansible-public RO\nsnmp-server community ansible-private RW\n"
}

TASK [include_role : ansible-network.config_manager] *******************************************************************************************************************************************************

TASK [ansible-network.config_manager : validate role spec] *************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.config_manager : create temp working directory] **************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : checkout scm project] ***********************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : discover the configuration file path and load it] *******************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : set the config_manager_file value based on config_manager_scm_file] *************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : remove temporary working dir] ***************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : load config file contents] ******************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.config_manager : validate config_manager_text is defined] ****************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [invoke network provider] *****************************************************************************************************************************************************************************

TASK [ansible-network.cisco_ios : initialize function] *****************************************************************************************************************************************************
included: /Users/wtome/Documents/customers/Marriott/network-poc/roles/ansible-network.cisco_ios/includes/init.yaml for rtr1, rtr2

TASK [ansible-network.cisco_ios : set role basic facts] ****************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : display the role version to stdout] **************************************************************************************************************************************
ok: [rtr2] => {
    "msg": "ansible_network.cisco_ios version is v2.6.2"
}
ok: [rtr1] => {
    "msg": "ansible_network.cisco_ios version is v2.6.2"
}

TASK [ansible-network.cisco_ios : validate ansible_network_os == 'ios'] ************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : validate ansible_connection == 'network_cli'] ****************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : Validate we have required installed version of dependent roles] **********************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : Validate we have required installed version of dependent roles from meta] ************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : validate config_manager_text is defined] *********************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : set ios checkpoint filename] *********************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : create checkpoint of current configuration] ******************************************************************************************************************************
included: /Users/wtome/Documents/customers/Marriott/network-poc/roles/ansible-network.cisco_ios/includes/checkpoint/create.yaml for rtr1, rtr2

TASK [ansible-network.cisco_ios : validate ios_checkpoint_filename is defined] *****************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : get current files on disk] ***********************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : remove old checkpoint file (if necessary)] *******************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : create a checkpoint of the current running-config] ***********************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : display message due to check mode] ***************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : replace current active configuration] ************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : merge with current active configuration] *********************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : load configuration using configure terminal] *****************************************************************************************************************************
included: /Users/wtome/Documents/customers/Marriott/network-poc/roles/ansible-network.cisco_ios/includes/configure/terminal.yaml for rtr1, rtr2

TASK [ansible-network.cisco_ios : enter configuration mode] ************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : load configuration lines into target device] *****************************************************************************************************************************
ok: [rtr2] => (item=snmp-server community ansible-public RO)
ok: [rtr1] => (item=snmp-server community ansible-public RO)
ok: [rtr1] => (item=snmp-server community ansible-private RW)
fatal: [rtr1]: FAILED! => {"msg": "missing required argument `command`"}
ok: [rtr2] => (item=snmp-server community ansible-private RW)
fatal: [rtr2]: FAILED! => {"msg": "missing required argument `command`"}

TASK [ansible-network.cisco_ios : exit configuration mode] *************************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : set host failed] *********************************************************************************************************************************************************
fatal: [rtr1]: FAILED! => {"changed": false, "msg": "error loading configuration lines"}
fatal: [rtr2]: FAILED! => {"changed": false, "msg": "error loading configuration lines"}

TASK [ansible-network.cisco_ios : display message] *********************************************************************************************************************************************************
ok: [rtr2] => {
    "msg": "error configuring device, starting rollback"
}
ok: [rtr1] => {
    "msg": "error configuring device, starting rollback"
}

TASK [ansible-network.cisco_ios : initiate configuration rollback] *****************************************************************************************************************************************
included: /Users/wtome/Documents/customers/Marriott/network-poc/roles/ansible-network.cisco_ios/includes/checkpoint/restore.yaml for rtr1, rtr2

TASK [ansible-network.cisco_ios : validate ios_checkpoint_filename is defined] *****************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : get current files on disk] ***********************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : verify checkpoint file exists] *******************************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : checkpoint configuration restore pre hook] *******************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : restore checkpoint configuration] ****************************************************************************************************************************************
ok: [rtr1]
ok: [rtr2]

TASK [ansible-network.cisco_ios : checkpoint configuration restore post hook] ******************************************************************************************************************************
skipping: [rtr1]
skipping: [rtr2]

TASK [ansible-network.cisco_ios : remove checkpoint file from remote device] *******************************************************************************************************************************
ok: [rtr2]
ok: [rtr1]

TASK [ansible-network.cisco_ios : display message] *********************************************************************************************************************************************************
ok: [rtr1] => {
    "msg": "successfully completed configuration rollback"
}
ok: [rtr2] => {
    "msg": "successfully completed configuration rollback"
}

TASK [ansible-network.cisco_ios : fail host due to config load error] **************************************************************************************************************************************
fatal: [rtr1]: FAILED! => {"changed": false, "msg": "error loading configuration onto target device"}
fatal: [rtr2]: FAILED! => {"changed": false, "msg": "error loading configuration onto target device"}

PLAY RECAP *************************************************************************************************************************************************************************************************
rtr1                       : ok=21   changed=0    unreachable=0    failed=3   
rtr2                       : ok=21   changed=0    unreachable=0    failed=3   

```

**Comment by willtome** [comment]
Here are the versions of the roles that I am using 

```ansible-galaxy list -p ./roles/
- ansible-network.config_manager, v2.6.2
- ansible-network.cisco_ios, v2.6.3
- ansible-network.aws, v2.6.0
- ansible-network.cloud_vpn, v2.6.3
- ansible-network.network-engine, v2.7.1
```

**Comment by gdpak** [comment]
@willtome Can you please try patching fix of PR #64 ?

**Comment by willtome** [comment]
@gdpak PR #64 fixed the issue when using network-engine v2.7.1. I found that the issue does not exist with network engine v2.6.5 so I'm wondering if the bug is here or really something that should be fixed in network-engine

**Comment by gdpak** [comment]
@willtome This need to be fixed here so that we don't call network-engine for config lines with just whitespace chars.
